framework/src/media/media_recorder.c

```diff
@@ -0,0 +1,348 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <media/media_recorder.h>
+#include <media/media_utils.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <fcntl.h>
+
+#define IS_SOCKET(fd) (fd > CONFIG_NFILE_DESCRIPTORS)
+#define RECORD_LOCK() do { pthread_mutex_lock(&g_rc.mutex_record); } while (false)
+#define RECORD_UNLOCK() do { pthread_mutex_unlock(&g_rc.mutex_record); } while (false)
+
+#define ST_PROTO_RECORD			0x04
```

Comment:
Hard to understand these value, please add some comment regarding these.
Also, please apply below things.
/****************************************************************************
 * Included Files
 ****************************************************************************/

/****************************************************************************
 * Pre-processor Definitions
 ****************************************************************************/

/****************************************************************************
 * Public Types
 ****************************************************************************/

/****************************************************************************
* Global Function Prototypes
****************************************************************************/